app/src/main/java/com/ataulm/artcollector/ArtCollectorApplication.kt

```diff
@@ -1,5 +1,20 @@
 package com.ataulm.artcollector
 
 import android.app.Application
+import android.content.Context
 
-class ArtCollectorApplication : Application()
+class ArtCollectorApplication : Application() {
+
+    private val component: ApplicationComponent by lazy {
+        DaggerApplicationComponent
+                .builder()
+                .application(this)
+                .build()
```

Comment:
this is essentially making the objects in the ApplicationComponent singleton per app process.

We don't need to manually inject anything from the ApplicationComponent into any of our classes (unlike DaggerPaintingsComponent which has an `inject(PaintingsActivity)` method) but we do need a reference to this component so that we can build the `PaintingsComponent` which depends on the object that this provides.